firestore/swift/firestore-smoketest/ViewController.swift

```diff
@@ -738,14 +739,20 @@ class ViewController: UIViewController {
         let docRef = db.collection("cities").document("BJ")
 
         docRef.getDocument { (document, error) in
-            if let city = document.flatMap({
-              $0.data().flatMap({ (data) in
-                return City(dictionary: data)
-              })
-            }) {
-                print("City: \(city)")
-            } else {
-                print("Document does not exist")
+            let result = Result {
+                try document.flatMap {
+                    try $0.data(as: City.self)
+                }
+            }
+            switch result {
+            case .success(let city):
+                if let city = city {
+                    print("City: \(city)")
+                } else {
+                    print("Document does not exist")
+                }
+            case .failure(let error):
+                print("Error decoding city: \(error)")
```

Comment:
I really wish this weren't so verbose.

Reply:
We can always go back to the
```swift
let city = (try? document.flatMap({ try $0.data(as: City.self) })).flatMap({ $0 })
```
if we're willing to ignore errors :)